src/wallet3/keyring.cpp

```diff
@@ -394,6 +391,58 @@ namespace wallet
       throw std::runtime_error("RCT signing went wrong -- verRctNonSemanticsSimple returned false");
   }
 
+  std::vector<crypto::public_key> Keyring::get_subaddress_spend_public_keys(uint32_t account, uint32_t begin, uint32_t end) {
+    if (begin <= end)
+      throw std::runtime_error("begin > end");
+
+    std::vector<crypto::public_key> pkeys;
+    pkeys.reserve(end - begin);
+    cryptonote::subaddress_index index = {account, begin};
+
+    ge_p3 p3;
+    ge_cached cached;
+    if (ge_frombytes_vartime(&p3, spend_public_key.data()) != 0)
+      throw std::runtime_error("ge_frombytes_vartime failed to convert spend public key");
+    ge_p3_to_cached(&cached, &p3);
+
+    for (uint32_t idx = begin; idx < end; ++idx)
+    {
+      index.minor = idx;
+      if (index.is_zero())
```

Comment:
I don't actually know if a subaddress index of {0,1+} is supported; would have to check existing implementation.

Reply:
yeah confident that the minor index is available for 0. CLI wallet shows this
```
Untagged accounts:
          Address               Balance      Unlocked balance        Batched Amount                 Label
 *       0 T6Tc4Z           0.000000000           0.000000000           0.000000000       Primary account
----------------------------------------------------------------------------------
          Total           0.000000000           0.000000000
Currently selected account: [0] Primary account
Tag: (No tag assigned)
Balance: 0.000000000, unlocked balance: 0.000000000
Background refresh thread started
[wallet T6Tc4Z]: address new
[wallet T6Tc4Z]: 1  TRsYca29pgPcM17jorNh2xRuNXqU8JzVYCeB93roHxHAbQnYmr3zDjSEfJjFug6nM8FGb7h29sYk7ZCMkNoWfcJS2E1acQf9G  (Untitled address) 
[wallet T6Tc4Z]: address all
[wallet T6Tc4Z]: 0  T6Tc4ZC4AH7ACNiTEJMvFha19zeL5ywtDMKQzkhe7H36GGtxhfT6c9bKVK6wcs7ube9gRJjroQxYUHJv8nVi5S6y24JTBQnXA  Primary address 
1  TRsYca29pgPcM17jorNh2xRuNXqU8JzVYCeB93roHxHAbQnYmr3zDjSEfJjFug6nM8FGb7h29sYk7ZCMkNoWfcJS2E1acQf9G  (Untitled address) 
```